src/appliance-onboarding-script/run.ps1

```diff
@@ -197,7 +198,14 @@ else
 $az_account_check_token = az account get-access-token
 if ($az_account_check_token -eq $null){
     setPathForAzCliCert -config $config
-    az login --use-device-code
+    if ($PSBoundParameters.ContainsKey('VmWareSPObjectID'))
```

Comment:
Can we use a separate flag to denote if this is for testing or for actual onboarding? 
It would be good to decouple the two. 
We would not need the object id soon, we would continue to need the identity then, for instance

Reply:
Done